extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerSketchFetcher.java

```diff
@@ -155,7 +152,17 @@ CompletableFuture<Either<Long, ClusterByPartitions>> inMemoryFullSketchMerging(
             mergedStatisticsCollector.clear();
```

Comment:
we can call this catch block only once by checking `partitionFuture.isDone`

Reply:
Changed